executor/aggfuncs/func_avg.go

```diff
@@ -56,7 +58,19 @@ func (e *baseAvgDecimal) AppendFinalResult2Chunk(sctx sessionctx.Context, pr Par
 	finalResult := new(types.MyDecimal)
 	err := types.DecimalDiv(&p.sum, decimalCount, finalResult, types.DivFracIncr)
 	if err != nil {
-		return errors.Trace(err)
+		return err
+	}
+	// Make the decimal be the result of type inferring.
+	frac := e.args[0].GetType().Decimal
```

Comment:
Let it be a member of `baseAvgDecimal`?

Reply:
@winoros It seems we do not need to do this:
1. AppendFinalResultResult2Chunk will only be called **once** during the executing.
2. `avgOriginal4DistinctDecimal` does not hold `baseAvgDecimal` as a member variable, so we'll only use `frac` here even define it as a member of `baseAvgDecimal`.